nexus-sdk/src/main/java/io/nexusrpc/handler/OperationContext.java

```diff
@@ -48,6 +50,11 @@ public Map<String, String> getHeaders() {
     return headers;
   }
 
+  /** Get a read only list of links attached to this context. */
+  public List<Link> getLinks() {
+    return Collections.unmodifiableList(links);
+  }
```

Comment:
Why not just return the mutable collection and remove add/set? If Go's primary collections were mutable/growable I'd suggest that there too.

Reply:
I don't want to expose the leak the internal collection, this is a pretty normal Java pattern https://javadoc.io/doc/io.opentelemetry/opentelemetry-api/latest/io/opentelemetry/api/trace/Span.html

Reply:
It's also a fairly normal pattern to just give them the mutable collection. I don't see any benefit to not doing so in this case even if some other libraries happen not to. And you actually are choosing to leak the internal collection (e.g. if it's changed later, so is this). If you _really_ didn't want to leak it, you'd copy it. But I won't make this blocking.

Reply:
Oh true I should copy it I will fix that